src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.tt

```diff
@@ -0,0 +1,124 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ import namespace="System" #>
+<#@ assembly name="System.Core" #>
+<#@ assembly name="System.Net.Primitives" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ output extension=".cs" #>
+<#@ include file="TlsCipherSuiteNameParser.ttinclude" #><#@
+    include file="TlsCipherSuite.cs" #>
+<# Array tlsEnumValues = typeof(TlsCipherSuite).GetEnumValues(); #>
+<# Array exchangeEnumValues = typeof(ExchangeAlgorithmTypeIndex).GetEnumValues(); #>
+<# Array cipherEnumValues = typeof(CipherAlgorithmTypeIndex).GetEnumValues(); #>
+<# Array cipherStrengthEnumValues = typeof(CipherAlgorithmStrengthIndex).GetEnumValues(); #>
+<# Array hashEnumValues = typeof(HashAlgorithmTypeIndex).GetEnumValues(); #>
+
+
+using System.Diagnostics;
+using System.Security.Authentication;
+
+namespace System.Net.Security
+{
+    internal partial struct SslConnectionInfo
+    {
+        private void MapCipherSuite(TlsCipherSuite cipherSuite)
+        {
+            TlsCipherSuite = cipherSuite;
+            KeyExchKeySize = 0;
+
+            ushort data = GetPackedData(cipherSuite);
+            Debug.Assert(data != 0, $"No mapping found for cipherSuite {cipherSuite}");
+
+            KeyExchangeAlg = (int)s_exchangeAlgorithmTypes[(data >> (16 - (4 * 1)) & 0xF)];
+            DataCipherAlg = (int)s_cipherEnumValues[(data >> (16 - (4 * 2)) & 0xF)];
+            DataKeySize = (int)s_cipherStrengthEnumValues[(data >> (16 - (4 * 3)) & 0xF)];
+            DataHashAlg = (int)s_hashEnumValues[(data >> (16 - (4 * 4)) & 0xF)];
+            DataHashKeySize = GetHashSize((HashAlgorithmType)DataHashAlg);
```

Comment:
This can be done using Span-based map, using `(data >> (16 - (4 * 4)) & 0xF)` as the index. We do not need a method.